Unix/doc/setup-kerberos-omi.md

```diff
@@ -0,0 +1,627 @@
+
+### Identity Mapping
+
+Linux, MacOS and Unix identify users via an integer user ID (uid). That user is then given privileges according to is membership 
+in various groups, which are identified by an integer group id.  The utility id will show both of these for a given user. 
+
+In Windows the user id is in the form of an SID, which is much longer, variable in length, and heirarchical. Groups also have SIDs.
+The user information provided by the Active Directory Controller is in the form of SIDs and must be mapped to the simpler 
+Linux/MacOS/Unix format. This is done in a very simple form by winbind or sssd on the client, but there are additional packages
+such as IdM or FreeIPA that perform this operation with considerably more sophistication.
+
+We have not found it necessary to use these.
+
+## Setting up the client machine
+
+Different Distros and Unix implementations have different tools to accomplish it, but to connect to AD, one must: 
+
+  - Set up automatic time sync with ADC/KDC. This is ussaly via the network time protocol, ntp.
+  - DNS set up so it sees the AD server and is part of the servers DNS domain.
+  - Hostname shorted to 15 characters or less, if needed. Fully qualified hostname by default.
+  - Kerberos and LDAP setup (sssd/realm or winbind and net ads)
+  - nsswitch.conf set to use ldap and sss or winbind
+  - PAM modules added to allow kerberos authentication.
+
+Winbind, which directly uses the samba service, was universally used as an active domain agent on linux until the
+most recent version of Centos/RH, SLES and Ubuntu.  It is being supplanted by the Red Hats
+System Security Services Daemon (sssd) which is more secure and handles identity mapping better.
+
+Winbind uses the utility *net ads* as a command line interface to join a domain, leave the domain, manage keytabs,
+add users and computers, etc.
+
+Both Sssd and winbind use "net ads" to perform a number of useful operations.
+
+MacOS uses its own suite of utilities based on a fork of the Heimdal kerberos implementation (http://www.h5l.org).
+
+Much of the process of preparing a machine to join an active domain is a matter of preparing. IF the server and client 
+machine is already set up right, then the process of joining usually goes very smoothly. If there are problems in the setup, 
+then debugging becomes difficult. This document will try to call out the things to check in advance.
+
+Due to the limitations of netbios which are still reflected in the system, it is a good idea to limit hostnames to 15 characters.
+
+Microsoft recommends this, and we have had some rather unpleasant issues around violating that rule.  The reason is that 
+the netbios host name (referred to in the domain controllers data as SAM is based on the netbios name. Deviation between the shortname and the SAM name can cause problems. It just isnt worth it.
+
+A link here.(https://technet.microsoft.com/en-us/library/cc726016(v=ws.10).aspx)
```

Comment:
A link here? Set this up right, first. Yeah, there's a link, but not a correct .md link. And I'm not sure what you're trying to say. "A link here." What does that mean?